ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/QueryDescription.java

```diff
@@ -50,7 +51,7 @@
   private final Map<KsqlHostInfoEntity, KsqlQueryStatus> ksqlHostQueryStatus;
   private final KsqlQueryType queryType;
   private final List<QueryError> queryErrors;
-  private final Set<StreamsTaskMetadata> tasksMetadata;
```

Comment:
Can this still be final?

Reply:
I ended up doing this way because I wanted to restrict how someone could interface with a QueryDescription object, they have to call the `updateTaskMetadata()` function to modify the set.

Reply:
would be a little more idiomatic to make the set final but not immutable and to have the getter make immutable copies.

Reply:
^ ok did that